src/library_gl.js

```diff
@@ -5312,7 +5312,7 @@ var LibraryGL = {
         if (!GL.currArrayBuffer) {
           GLImmediate.firstVertex = end ? start : TOTAL_MEMORY; // if we don't know the start, set an invalid value and we will calculate it later from the indices
           GLImmediate.lastVertex = end ? end+1 : 0;
-          GLImmediate.vertexData = {{{ makeHEAPView('F32', 'GLImmediate.vertexPointer', '(end ? GLImmediate.vertexPointer + (end+1)*GLImmediate.stride : TOTAL_MEMORY)') }}}; // XXX assuming float
+          GLImmediate.vertexData = HEAPF32.subarray(GLImmediate.vertexPointer >> 2, end ? (GLImmediate.vertexPointer + (end+1)*GLImmediate.stride) >> 2 : undefined); // XXX assuming float
```

Comment:
In general, `makeHEAPView` is not needed, we can just write `subarray`s.

Note that `heapAndOffset` does more: it returns `subarray, offset`, as a combination. This particular line cant use that, due to the indexing (see the `undefined`).

In split heap mode, each malloc goes in a single heap slice, so an allocation cannot span two chunks. And `HEAPF32` are just "fake" objects that exist in order to provide `.subarray` etc., which they do by providing a slice of the proper chunk. This is definitely not obvious, I started to write some docs now at https://github.com/kripken/emscripten/wiki/Split-Memory